### ISSUE TITLE
Feature Request: Add another abstraction layer of underlying "quadrants" for object placement

### ISSUE THREAD

**Issue by finnje** [mingrammer/diagrams#762]
When you are trying to represent something large/complex the visualization part of diagrams has a hard time displaying in a consumable format.  If there was another layer where you could specify an area or quadrant on a canvas to represent subsets of objects it would allow you to "help" organize what gets diagrammed in a more useful way.

**Comment by clayms** [comment]
What about Clusters?  
Please provide a sketch to show what you are describing.

**Comment by PabloArisi00** [comment]
Hello I love this product! I agree with @finnje , it would be ideal to be able to build a graph of this complexity.
https://d1.awsstatic.com/architecture-diagrams/ArchitectureDiagrams/mobile-web-serverless-RA.pdf?did=wp_card&trk=wp_card
https://d1.awsstatic.com/architecture-diagrams/ArchitectureDiagrams/direct-connect-traffic-flow-local-zone-ra.pdf?did=wp_card&trk=wp_card

Thank you very much!

**Comment by TechDH** [comment]
I agree with these comments. My biggest issue is problems with the auto placement of nodes and clusters.  For big diagrams with lots of nodes inside the same cluster it stays on the same plane and does not wrap. I use diagrams in documentation and presentations so I need to control the width and height of the diagram but there is lack of control. It would be good to have control of layout better and combine the LR or TB  at multiple levels not just the top might help. For example if I have 3 clusters one of which is really big based on what's nested inside, it might render in LR with the 3 clusters in order from left to right or have one at the bottom when a more logical view might be to have the smaller clusters on the left or at the top. I don't see anyway to control this and to override the default placement and wrap a large number of nodes within a cluster to a defined maximum width. Just a few examples of the lack of control over cluster and node placement in larger diagrams.

**Comment by clayms** [comment]
@TechDH 
See the code in this response for "wraping" the nodes by setting the number of nodes per row, i.e. the number of columns. 
https://github.com/mingrammer/diagrams/issues/290#issuecomment-719983090
